pkg/staticnetworkconfig/generator.go

```diff
@@ -122,7 +122,8 @@ func (s *StaticNetworkConfigGenerator) executeNMStatectl(ctx context.Context, ho
 	if retCode != 0 {
 		msg := fmt.Sprintf("<nmstatectl gc> failed, errorCode %d, stderr %s, input yaml <%s>", retCode, stderr, hostYAML)
 		s.log.Errorf("%s", msg)
-		return "", fmt.Errorf("%s", msg)
+		returnMsg := fmt.Sprintf("Failed to execute nmstatectl gc, check the value provided for 'network_yaml': %s", hostYAML)
+		return "", fmt.Errorf("%s", returnMsg)
```

Comment:
I think this can be simplified now that we're not using the same value for the return and log. For example we could do:

```
s.log.Errorf("<nmstatectl gc> failed, errorCode %d, stderr %s, input yaml <%s>", retCode, stderr, hostYAML)
return "", fmt.Errorf("Failed to execute nmstatectl gc, check the value provided for 'network_yaml': %s", hostYAML)
```

Also, are we pretty sure this will be caused by an invalid input?
If so, we should say that more clearly. Maybe something like "Failed to execute nmstatectl gc, invalid input yaml: %s".

Lastly, the convention for errors is to not capitalize the first word of the message so that it can be wrapped and look reasonable.

So eventually this would be something like `Update infraEnv failed: failed to execute...` rather than `Update infraEnv failed: Failed to exectute...`

Reply:
Thank you for pointing out the simplification and the note about the convention! I'll keep that in mind for next time too.

As far as errors, for the most part, I've seen the invalid input error. However, I added a unit test for this and there was an error that says the `nmstatectl` command was missing. I'm not sure if this would be relevant in production, what do you think?

Reply:
> error that says the nmstatectl command was missing. I'm not sure if this would be relevant in production

We definitely need the command to be present for any of this to work, so I don't think we'd run into that in prod.
That said, I don't have strong opinions on the wording so what you have is probably fine.

Reply:
Great that's good to know! Thanks Nick!